### DISCUSSION TITLE
Determinant for a matrix with polynomial entries

### DISCUSSION THREAD

**Discussion by adhalanay** [oscar-system/Oscar.jl#3629] [Q&A] [answered]
Hello.

 I have a matrix of type 3x3 Matrix{QQMPolyRingElem} and I am trying to compute its determinant but the `det` function seems to not be defined for this kind of arguments. Is there some other function?

**Comment by joschmitt** [comment, marked as answer]
We have our own "matrix type" in OSCAR (if you wonder why, see the question "Why do you have your own matrix types, and why do they not support the exact same commands as Julia matrices?" in the [FAQ](https://docs.oscar-system.org/stable/General/faq/)). So, you have to convert your Julia `Matrix` into an OSCAR one:
```
julia> R, (x, y) = QQ["x", "y"];

julia> M = [x y; y x] # <- This is a Julia Matrix, the OSCAR functionality usually doesn't work
2×2 Matrix{QQMPolyRingElem}:
 x  y
 y  x

julia> N = matrix(R, M) # This is now an OSCAR matrix
[x   y]
[y   x]

julia> det(N) # Now `det` works
x^2 - y^2
```
